src/openrct2/ride/Ride.cpp

```diff
@@ -4833,91 +4833,139 @@ uint8_t ride_entry_get_vehicle_at_position(int32_t rideEntryIndex, int32_t numCa
+            { SpriteGroupType::Slopes12, SpritePrecision::Sprites4 },      // TRACK_RAPIDS
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60, SpritePrecision::Sprites4,
+              SpriteGroupType::Slopes75, SpritePrecision::Sprites4, SpriteGroupType::Slopes90,
+              SpritePrecision::Sprites4 }, // TRACK_HALF_LOOP_UNINVERTED
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60, SpritePrecision::Sprites4,
+              SpriteGroupType::Slopes75, SpritePrecision::Sprites4, SpriteGroupType::Slopes90, SpritePrecision::Sprites4,
+              SpriteGroupType::SlopesLoop, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 },                             // TRACK_HALF_LOOP_INVERTED
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_WATERFALL
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_WHIRLPOOL
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60,
+              SpritePrecision::Sprites4 }, // TRACK_BRAKE_FOR_DROP
+            { SpriteGroupType::Corkscrews, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 }, // TRACK_CORKSCREW_UNINVERTED
+            { SpriteGroupType::Corkscrews, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 }, // TRACK_CORKSCREW_INVERTED
+            { SpriteGroupType::Slopes12, SpritePrecision::Sprites4, SpriteGroupType::Slopes25,
+              SpritePrecision::Sprites4 },                             // TRACK_HEARTLINE_TRANSFER
+            {},                                                        // TRACK_MINI_GOLF_HOLE
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_ROTATION_CONTROL_TOGGLE
+            { SpriteGroupType::Slopes60, SpritePrecision::Sprites4 },  // TRACK_SLOPE_STEEP_UP
+        };
 
     // Only check default vehicle; it's assumed the others will have correct sprites if this one does (I've yet to find an
     // exception, at least)
     auto supportedPieces = std::numeric_limits<uint64_t>::max();
     auto defaultVehicle = rideEntry->GetDefaultVehicle();
     if (defaultVehicle != nullptr)
     {
-        const auto defaultSpriteFlags = defaultVehicle->sprite_flags;
         for (size_t i = 0; i < std::size(trackPieceRequiredSprites); i++)
         {
-            if ((defaultSpriteFlags & trackPieceRequiredSprites[i]) != trackPieceRequiredSprites[i])
+            for (auto& group : trackPieceRequiredSprites[i])
             {
-                supportedPieces &= ~(1ULL << i);
+                auto precision = defaultVehicle->SpriteGroups[static_cast<uint8_t>(group.VehicleSpriteGroup)].spritePrecision;
+                if (precision < group.MinPrecision && precision > SpritePrecision::None)
```

Comment:
This is the actual bug here. ` && precision > SpritePrecision::None` delete this as it makes no sense. We are trying to find the sprite group that the track does not have which is most likely a `None`